File: 프로그래머스/unrated/181918. 배열 만들기 4/배열 만들기 4.js
function solution(arr) {
    let stk = [];
    let idx = 0;
    
    
    while(idx < arr.length){
        if(stk.length === 0){
          stk.push(arr[idx]);
          idx++;
        }else{
            if(stk[stk.length-1] < arr[idx]){
              stk.push(arr[idx]);
              idx++;
          }
            else{
              stk.pop();
          }
        }
    }
    return stk;
}

File: 프로그래머스/0/181918. 배열 만들기 4/배열 만들기 4.py
def solution(arr):
    stk = []
    i = 0
    while(i < len(arr)):
        if len(stk) == 0:
            stk.append(arr[i])
        elif stk[-1] < arr[i]:
            stk.append(arr[i])
        elif stk[-1] >= arr[i]:
            stk.pop()
            continue
        i+=1
            
    return stk